pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -128,3 +183,139 @@ func localJWTForKey(id string, privateKkeyPem string) (string, error) {
 
 	return tokenString, nil
 }
+
+func initK8sClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var err error
+	var config *rest.Config
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	k8sclientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sclientset, err
+}
+
+func (a *AuthConfig) createOrUpdateAuthTokenSecret(k8sclientset kubernetes.Interface) error {
+	// check if secret exists
+	retrievedSecret, err := k8sclientset.CoreV1().Secrets(authTokenSecretNamespace).Get(context.Background(), authTokenSecretName, metav1.GetOptions{})
+
+	// if the secret does not exist
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Create the secret with the new JWT token having an exp claim of 48 hours
+			// i.e. the token will be valid for next 48 hours
+			err = createSecret(k8sclientset, a.AgentAuthToken)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		// Other errors while trying to get the secret
+		return fmt.Errorf("failed to get required auth token secret from the cluster: %w. Re-run 'add-nodes' command to create new image files(ISO/PXE files)", err)
+	}
+
+	// if the secret exists in the cluster, get the token
+	retrievedToken, err := getExistingAuthToken(retrievedSecret)
+	if err != nil {
+		return err
+	}
+	expiryTime, err := ParseExpirationFromToken(retrievedToken)
+	if err != nil {
+		return err
+	}
+	// Calculate 24 hours before the expiration time
+	thresholdTime := expiryTime.Add(-24 * time.Hour)
+
+	// Check if current time is past the thresholdTime time of 24 hours
+	if time.Now().After(thresholdTime) {
+		// update the secret in the cluster with a new token from asset store
+		authToken := a.AgentAuthToken
+		err = updateSecret(k8sclientset, retrievedSecret, authToken)
```

Comment:
I think that by refactoring this func as a normal type method the signature could be greatly simplified